unstructured_ingest/v2/processes/connectors/github.py

```diff
@@ -0,0 +1,348 @@
+
+        Returns:
+            content_file: The file content object.
+
+        Raises:
+            UnknownObjectException: If the file does not exist in the repository.
+        """
+        from github.GithubException import UnknownObjectException
+
+        try:
+            logger.info(f"Fetching file from path: {path!r}")
+            content_file = self.connection_config.get_repo().get_contents(path)
+        except UnknownObjectException:
+            logger.error(f"File doesn't exist: {self.connection_config.url}/{path}")
+            raise UnknownObjectException
+
+        return content_file
+
+    @SourceConnectionNetworkError.wrap
+    @requires_dependencies(["httpx", "github"], extras="github")
+    async def _fetch_content(self, content_file):
+        """Asynchronously retrieves the content of a file, handling large files via direct download.
+
+        Args:
+            content_file: The file object from the GitHub API.
+
+        Returns:
+            bytes: The content of the file as bytes.
+
+        Raises:
+            UnknownObjectException: If the download or content retrieval fails.
+        """
+        import httpx
+        from github.GithubException import UnknownObjectException
+
+        contents = b""
+        async with httpx.AsyncClient() as client:
+            if not content_file.content and content_file.encoding == "none" and content_file.size:
+                logger.info(
+                    "File too large for the GitHub API, using direct download link instead."
+                )
+                try:
+                    response = await client.get(content_file.download_url, timeout=10.0)
+                    response.raise_for_status()
+                except (httpx.HTTPStatusError, httpx.RequestError) as e:
+                    logger.error(f"Failed to download {content_file.download_url}: {e}")
+                    raise UnknownObjectException
+
+                contents = response.content
+            else:
+                contents = content_file.decoded_content
+        return contents
+
+    async def _fetch_and_write(self, path: str, download_path: str) -> None:
+        """Fetches a file from GitHub and writes its content to the specified local path.
+
+        Args:
+            path (str): The path to the file in the repository.
+            download_path (str): The local path to save the downloaded file.
+
+        Raises:
+            ValueError: If the file content could not be retrieved.
+        """
+        content_file = self._fetch_file(path)
+        contents = await self._fetch_content(content_file)
+        if contents is None:
+            raise ValueError(
+                f"Failed to retrieve file from repo "
+                f"{self.connector_config.url}/{self.path}. Check logs",
+            )
+
+        logger.info(f"Writing download file to path: {download_path!r}")
+        with open(download_path, "wb") as f:
```

Comment:
When using a `Path`:
```suggestion
        with download_path.open("wb") as f:
```

Reply:
Done.